tests/cephadm/test_verify_cluster_health_after_reboot.py

```diff
@@ -0,0 +1,56 @@
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.utils import get_service_id, reboot_node
+
+
+class RebootFailed(Exception):
+    pass
+
+
+class ClusterNotHealthy(Exception):
+    pass
+
+
+class SystemctlFailed(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """Verify rebooting a node doesn't affect the ceph health
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    action = config.get("action")
+    mon_node = ceph_cluster.get_nodes("mon")[0]
+
+    if action == "node-reboot":
+        for node in ceph_cluster.get_nodes():
+            status = reboot_node(node)
+            if not status:
+                raise RebootFailed(f"Node {node.hostname} failed to reboot.")
+
+            # Verify the reboot doesn't affect the cluster health
+            if CephAdm(mon_node).ceph.health() != "HEALTH_OK":
+                raise ClusterNotHealthy(
+                    f"Ceph cluster not healthy after rebooting {node.hostname}"
+                )
+
+    elif action == "service-state":
+        # Get the SERVICE_ID of the mon service
+        mon_service = get_service_id(mon_node, "mon")
+
+        for state in ["start", "stop", "restart"]:
+            _, err = mon_node.exec_command(
```

Comment:
Same as `get_service_id` we can have `set_service_state`

Reply:
Noted Vaibhav. Will move it to utils